net: hostname: Add a function to set the hostname postfix as is

VineetaNarkhede-eaton opened the pull request:
Fixes: #72363

The existing function to set the postfix is converting postfix string to hexadecimal. Adding a new function to handle a use case where the provided postfix should be used as is without any conversion.

prayassamriya commented on subsys/net/hostname.c:
```diff
@@ -97,6 +97,39 @@ int net_hostname_set_postfix(const uint8_t *hostname_postfix,
+#if !defined(CONFIG_NET_HOSTNAME_UNIQUE_UPDATE)
+	postfix_set = true;
+#endif
+
```
Do we need to trigger net event like done in other function?
 what is its purpose?

VineetaNarkhede-eaton replied:
trigger_net_event() is to trigger the event upon hostname change. Added the same into new function.

MarkVerheyen-eaton commented on subsys/net/hostname.c:
We need the associated documentation with this. What does it do, why does it do it, how do we use it?

VineetaNarkhede-eaton replied:
Updated the readme file and header file.

rlubos commented on include/zephyr/net/hostname.h:
```diff
@@ -108,8 +109,31 @@ int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 static inline int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 					   int postfix_len)
 {
-	ARG_UNUSED(hostname_postfix);
```
Please don't remove `ARG_UNUSED()` macros here. And add them to the new function too.

VineetaNarkhede-eaton replied:
Fixed.

rlubos commented:
Regarding the compliance report:
```
5b9cdbb0a5569f494a6256000a28755f31005bd6: author email (vineeta <vineetasnarkhede@eaton.com>) needs to match one of the signed-off-by entries.
5b9cdbb0a5569f494a6256000a28755f31005bd6: author email (vineeta <vineetasnarkhede@eaton.com>) does not follow the syntax: First Last <email>.
```
You likely need to update your user.name and user.email git configs, and update the commit (`git commit --amend --reset-author`). They need to match the ones in the signoff entry.

There are also some trailing spaces, see the report.

VineetaNarkhede-eaton committed:
net: hostname: Add a function to set the hostname postfix as is.

Fixes: #72363

The existing function to set the postfix is converting postfix
string to hexadecimal. Adding a new function to handle a use case
where the provided postfix should be used as is without any conversion.

Signed-off-by: Vineeta S Narkhede <VineetaSNarkhede@Eaton.com>

VineetaNarkhede-eaton commented:
@rlubos @jukkar Could you please help me to merge the PR.

rlubos commented:
@VineetaNarkhede-eaton We're in a feature freeze, this will be merged after 3.7 release.
